spec/helpers/grades_helper_spec.rb

```diff
@@ -0,0 +1,196 @@
+      params[:id] = @assignment2.id
+      result = has_team_and_metareview?
+      expect(result).to be == {has_team: true, has_metareview: true, true_num: 2}
+    end
+  end
+
+  describe 'participant' do
+    it 'should return the correct particpant' do
+      new_participant = create(:participant)
+      params[:id] = new_participant.id
+      result = participant
+      expect(result).to eq(new_participant)
+    end
+  end
+
+  describe 'rscore_X' do
+    it 'should return records of each review type if available' do
+      params[:id] = @new_participant.id
+      expect(rscore_review).to_not be_nil
+      expect(rscore_metareview).to_not be_nil
+      expect(rscore_feedback).to_not eq(nil)
+      expect(rscore_teammate).to_not eq(nil)
+    end
+    it 'should return nil if review types are not available' do
+      params[:id] = @participant.id
+      expect(rscore_review).to be_nil
+      expect(rscore_metareview).to be_nil
+      expect(rscore_feedback).to be_nil
+      expect(rscore_teammate).to be_nil
+    end
+  end
+
+  describe 'p_total_score' do
+    it 'should return the grade if available' do
+      graded_participant = create(:participant, grade: 90)
+      create(:assignment_questionnaire, user_id: graded_participant.id, questionnaire: @questionnaire)
+      @questions[@questionnaire.symbol] = @questionnaire.questions
+      params[:id] = graded_participant.id
+      expect(p_total_score).to eq(90)
+    end
+    it 'should return :total_score if no grade is available' do
+      create(:assignment_questionnaire, user_id: @new_participant.id, questionnaire: @questionnaire)
+      @questions[@questionnaire.symbol] = @questionnaire.questions
+      params[:id] = @new_participant.id
+      expect(p_total_score).to eq(0)
+    end
+  end
+
+  describe 'p_title' do
+    it 'should return a title when the participant has a grade' do
+      params[:id] = @new_participant.id
+      @new_participant.grade = 90
+      @new_participant.save
+      expect(p_title).to eq('A score in blue indicates that the value was overwritten by the instructor or teaching assistant.')
+    end
+    it 'should return nil when the participant has no grade' do
+      params[:id] = @new_participant.id
+      expect(p_title).to eq(nil)
+    end
+  end
+
+  describe 'get_css_style_for_X_reputation' do
+    hamer_input = [-0.1, 0, 0.5, 1, 1.5, 2, 2.1]
+    lauw_input = [-0.1, 0, 0.2, 0.4, 0.6, 0.8, 0.9]
+    output = %w(c1 c1 c2 c2 c3 c4 c5)
+    it 'should return correct css for hamer reputations' do
+      hamer_input.each_with_index do |e, i|
+        expect(get_css_style_for_hamer_reputation(e)).to eq(output[i])
+      end
+    end
+    it 'should return correct css for luaw reputations' do
+      lauw_input.each_with_index do |e, i|
+        expect(get_css_style_for_lauw_reputation(e)).to eq(output[i])
+      end
+    end
+  end
+end
+
+#########################
+# Functional Cases
+#########################
+describe GradesHelper, type: :feature do
```

Comment:
You are welcome to write functional (feature) tests. But it is better to put tests to `spec/features` folder.